File: atcoder/abc192/abc192_c.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;

public class Main{
    public static void main(String[] args) throws IOException {
        BufferedReader bf = new BufferedReader(new InputStreamReader(System.in));
        String[] data = bf.readLine().split(" ");

        int k = Integer.parseInt(data[1]);
        for (int i = 0; i < k; i++) {
            String a = data[0];
            char[] chars = a.toCharArray();
            Arrays.sort(chars);
            a = new String(chars);
            StringBuilder b = new StringBuilder(new String(chars));
            b.reverse();

            data[0] = String.valueOf((Integer.parseInt(b.toString()) - Integer.parseInt(a)));
        }

        System.out.println(data[0]);
    }
}

File: atcoder/abc192/abc192_c.kt
package atcoder

import java.io.BufferedReader
import java.io.InputStreamReader

fun main() {
    val In = BufferedReader(InputStreamReader(System.`in`))
    val input = In.readLine().split(" ").map{ it.toInt()}
    fun g1(n : String) : Int {
        var array = n.toCharArray()
        array.sortDescending()
        return array.joinToString("").toInt()
    }
    fun g2(n : String) : Int {
        var array = n.toCharArray()
        array.sort()
        return array.joinToString("").toInt()
    }
    fun f(number : String) : Int {
        return g1(number) - g2(number)
    }
    val a = IntArray(input[1]+1)
    a[0] = input[0]
    for(i in 1..input[1]) {
        a[i] = f(a[i-1].toString())
    }
    println(a[input[1]])
}